src/MOLFiniteDifference/MOL_discretization.jl

```diff
@@ -281,16 +284,17 @@ function SciMLBase.symbolic_discretize(pdesys::ModelingToolkit.PDESystem,discret
                 central_deriv_rules_spherical = [Differential(s)(s^2*Differential(s)(u))/s^2 => central_deriv_spherical(II,j,k) 
                                                 for (j,s) in enumerate(nottime), (k,u) in enumerate(depvars)]
                 
+                # Val rules ############################################################
                 valrules = vcat([depvars[k] => depvarsdisc[k][II] for k in 1:length(depvars)],
                                 [nottime[j] => grid[j][II[j]] for j in 1:nspace])
-            
-                # TODO: upwind rules needs interpolation into `@rule`
+
+                # Upwind rules #########################################################
                 forward_weights(II,j) = DiffEqOperators.calculate_weights(discretization.upwind_order, 0.0, grid[j][[II[j],II[j]+1]])
                 reverse_weights(II,j) = DiffEqOperators.calculate_weights(discretization.upwind_order, 0.0, grid[j][[II[j]-1,II[j]]])
-                # upwinding_rules = [@rule(*(~~a,$(Differential(nottime[j]))(u),~~b) => IfElse.ifelse(*(~~a..., ~~b...,)>0,
-                #                         *(~~a..., ~~b..., dot(reverse_weights(II,j),depvars[k][central_neighbor_idxs(II,j)[1:2]])),
-                #                         *(~~a..., ~~b..., dot(forward_weights(II,j),depvars[k][central_neighbor_idxs(II,j)[2:3]]))))
-                #                         for j in 1:nspace, k in 1:length(pdesys.depvars)]
+                upwinding_rules_tmp = [@rule(*(~~a,$(Differential(iv)),~~b) => IfElse.ifelse(*(~~a..., ~~b...,)>0,
```

Comment:
The ifelse will work if `a` and `b` can be evaluated. But what if they are dependent variables (e.g. velocity in Navier-Stokes)?